pkg/client/userd/grpc.go

```diff
@@ -498,22 +501,56 @@ func (s *Service) ListCommands(ctx context.Context, _ *empty.Empty) (groups *rpc
 	return
 }
 
-func stdinPump(ctx context.Context, cmdStream rpc.Connector_RunCommandServer, wr io.Writer) {
-	for ctx.Err() == nil {
-		cr, err := cmdStream.Recv()
-		if err != nil {
-			if err != io.EOF && ctx.Err() == nil {
-				dlog.Errorf(ctx, "command failed to read stdin: %v", err)
-			}
-			break
+func stdinPump(ctx context.Context, cmdStream rpc.Connector_RunCommandServer) (context.Context, io.Reader, error) {
+	var wr io.WriteCloser
+	var rd io.Reader
+	isPTY := true
+	if runtime.GOOS == "windows" {
+		pi := ioutils.NewBytesPipe()
+		wr = pi
+		rd = pi
+		isPTY = false
+	} else {
+		// Use a PTY on unix systems.
+		var err error
+		if wr, rd, err = pty.Open(); err != nil {
```

Comment:
should we log this error?
another option could be to open the pty and doing an error check before launching this func as goroutine, this way we also avoid the external process exiting because this silently failed.

Reply:
Good catch! I meant for this error to be returned so and propagated to the user.